hudi-common/src/main/java/org/apache/hudi/common/util/Option.java

```diff
@@ -76,6 +76,10 @@ public static <T> Option<T> of(T value) {
     return new Option<>(value);
   }
 
+  public static <T> Option<T> ofNullable(Supplier<? extends T> value) {
+    return null == value ? empty() : ofNullable(value.get());
+  }
```

Comment:
this API looks weird: why would a Supplier be `null` ? usually you use supplier to supply a fallback value when input is null, e.g.,`orElse(Supplier<? extends T> supplier)`

Reply:
@xushiyan It's my mistake. Suppliers are supported in java9+, so I want to add it, but I lack consideration when defining methods.

Reply:
@cdmikechen no worries